### PR TITLE
Validate mkdocs.yml config on values that we manipulate

### PR DESCRIPTION
**Pull request by humitos**
Make sure that the values we will manipulate from the mkdocs.yml filethe co rrect type we need them to be. If we found a problem with them, we report a proper message to user to debug it by themselves.

captura de pantalla_2019-01-16_12-03-32

### CODE REVIEW

File: readthedocs/doc_builder/exceptions.py [hunk 1]
```diff
@@ -59,3 +59,13 @@ class MkDocsYAMLParseError(BuildEnvironmentError):
     GENERIC_WITH_PARSE_EXCEPTION = ugettext_noop(
         'Problem parsing MkDocs YAML configuration. {exception}',
     )
+
+    INVALID_DOCS_DIR_CONFIG = ugettext_noop(
+        'The "docs_dir" config from your MkDocs YAML config file has to be a '
+        'string with relative or absolute path.',
```
Comment (ericholscher) [inline readthedocs/doc_builder/exceptions.py]: It feels weird to quote this, we should just say it.

File: readthedocs/doc_builder/exceptions.py [hunk 2]
```diff
@@ -59,3 +59,13 @@ class MkDocsYAMLParseError(BuildEnvironmentError):
...
+    )
+
+    INVALID_EXTRA_CONFIG = ugettext_noop(
+        'The "{config}" config from your MkDocs YAML config file has to be a '
+        'a list of relative paths.',
```
Comment (ericholscher) [inline readthedocs/doc_builder/exceptions.py]: Same here.

File: readthedocs/doc_builder/exceptions.py (hunk 1 above)
Comment (ericholscher) [reply readthedocs/doc_builder/exceptions.py]: Nevermind, I read this wrong.

**Comment by humitos** [comment]
> Looks good, after tests pass. +1

The test that failed helped me to find a bug in my code ;)

Also, I added tests for the case of invalid configs also. I'll merge once test pass on Travis.

**Pull request merged**